ui/components/component-library/tag-url/tag-url.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import classnames from 'classnames';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderColor,
+  BorderRadius,
+  Display,
+  IconColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import {
+  AvatarFavicon,
+  ButtonLink,
+  Box,
+  IconName,
+  Icon,
+  IconSize,
+  Text,
+  ButtonLinkSize,
+} from '..';
+import { BoxProps, PolymorphicRef } from '../box';
+import { TagUrlComponent, TagUrlProps } from './tag-url.types';
+
+export const TagUrl: TagUrlComponent = React.forwardRef(
+  <C extends React.ElementType = 'div'>(
+    {
+      label,
+      labelProps,
+      actionButtonLabel,
+      actionButtonProps,
+      src,
+      showLockIcon,
+      children,
+      avatarFaviconProps,
+      lockIconProps,
+      className = '',
+      ...props
+    }: TagUrlProps<C>,
+    ref?: PolymorphicRef<C>,
+  ) => {
+    return (
+      <Box
+        className={classnames('mm-tag-url', className)}
+        ref={ref}
+        backgroundColor={BackgroundColor.backgroundDefault}
+        borderColor={BorderColor.borderDefault}
+        borderWidth={1}
+        alignItems={AlignItems.center}
+        paddingLeft={2}
+        paddingRight={4}
+        gap={2}
+        borderRadius={BorderRadius.pill}
+        display={Display.Flex}
+        {...(props as BoxProps<C>)}
+      >
+        <AvatarFavicon src={src} name={label} {...avatarFaviconProps}>
+          {children}
+        </AvatarFavicon>
```

Comment:
the TS error was due to not passing `children` here. 
@georgewrmarshall  Could you please check this when you do a full review and confirm if it is acceptable? Thank you!

Reply:
Hey @dhruvv173, I think the move here is to omit it from `AvatarFavicon` as a prop. I've made that update everything else looks pretty good 💯